app/lib/actions/pulp/repository/sync.rb

```diff
@@ -66,7 +66,8 @@ def run_progress_weight
         end
 
         def presenter
-          repo = ::Katello::Repository.find(input['repo_id'])
+          repo = ::Katello::Repository.find(input['repo_id']) if input['repo_id']
+          repo = ::Katello::Repository.where(:pulp_id => input['pulp_id']).first if (!repo && input['pulp_id'])
```

Comment:
nitpic: you could use `repo ||=`  and avoid the !repo check

I would also probably add a comment to explain that older sync tasks could use the pulp_id (otherwise it might get removed in the future.  Otherwise, looks good!